src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaCloudformation.tsx

```diff
@@ -1,30 +1,79 @@
 import React from 'react';
 import styled from '@emotion/styled';
+import * as qs from 'query-string';
 
 import {addErrorMessage, addLoadingMessage} from 'app/actionCreators/indicator';
-import Link from 'app/components/links/link';
+import ExternalLink from 'app/components/links/externalLink';
 import {t, tct} from 'app/locale';
+import {uniqueId} from 'app/utils/guid';
 import Form from 'app/views/settings/components/forms/form';
 import JsonForm from 'app/views/settings/components/forms/jsonForm';
 import FormModel from 'app/views/settings/components/forms/model';
 import {JsonFormObject} from 'app/views/settings/components/forms/type';
 
+// let the browser generate and store the external ID
+// this way the same user always has the same external ID if they restart the pipeline
+const ID_NAME = 'AWS_EXTERNAL_ID';
+const getAwsExternalId = () => {
+  let awsExternalId = window.localStorage.getItem(ID_NAME);
+  if (!awsExternalId) {
+    awsExternalId = uniqueId();
+    window.localStorage.setItem(ID_NAME, awsExternalId);
+  }
+  return awsExternalId;
+};
+
 type Props = {
-  cloudformationUrl: string;
-  awsExternalId: string;
+  baseCloudformationUrl: string;
+  templateUrl: string;
+  stackName: string;
+  arn?: string;
+  error?: string;
 };
 
 export default class AwsLambdaCloudformation extends React.Component<Props> {
-  model = new FormModel({apiOptions: {baseUrl: window.location.origin}});
+  componentDidMount() {
+    // show the error if we have it
+    const {error} = this.props;
+    if (error) {
+      addErrorMessage(error, {duration: 10000});
+    }
+  }
+  model = new FormModel();
   get initialData() {
+    const {arn} = this.props;
+    const awsExternalId = getAwsExternalId();
     return {
-      awsExternalId: this.props.awsExternalId,
+      awsExternalId,
+      arn,
     };
   }
-  handlePreSubmit = () => addLoadingMessage(t('Submitting\u2026'));
-  handleSubmitError = () => addErrorMessage(t('Unexpected error ocurred!'));
+  get cloudformationUrl() {
+    // generarate the cloudformation URL using the params we get from the server
+    // and the external id we generate
+    const {baseCloudformationUrl, templateUrl, stackName} = this.props;
+    const awsExternalId = getAwsExternalId();
+    const query = qs.stringify({
+      templateURL: templateUrl,
+      stackName,
+      param_ExternalId: awsExternalId,
+    });
+    return `${baseCloudformationUrl}?${query}`;
+  }
+  handleSubmit = (data: any) => {
+    addLoadingMessage(t('Submitting\u2026'));
+    this.model.setFormSaving();
+    const {
+      location: {origin},
+    } = window;
+    // redirect to the extensions endpoint with the form fields as query params
+    // this is needed so we don't restart the pipeline loading from the original
+    // OrganizationIntegrationSetupView route
+    const newUrl = `${origin}/extensions/aws_lambda/setup/?${qs.stringify(data)}`;
+    window.location.replace(newUrl);
```

Comment:
@scefali this view is in the popup, right? Doing `window.location.replace()` will prevent the user from using the back button. Maybe we can use `window.location.assign()`? Or also can't we use React Router?

Reply:
@NisanthanNanthakumar `window.location.assign()` works, I'll switch to that